src/commands/compile/spec/compile.spec.ts

```diff
@@ -81,3 +86,63 @@ describe('sasjs compile', () => {
     done()
   })
 })
+
+describe('sasjs compile', () => {
+  let appName: string
+  let target: Target
+
+  afterEach(async (done) => {
+    await removeTestApp(__dirname, appName)
+    jest.clearAllMocks()
+
+    done()
+  })
+
+  describe('job', () => {
+    beforeEach(async (done) => {
+      appName = `cli-tests-compile-${generateTimestamp()}`
+      await createTestJobsApp(__dirname, appName)
+      jest.spyOn(compileJobFile, 'compileJobFile')
+      done()
+    })
+
+    it('should compile single file', async (done) => {
+      await expect(
+        compileSingleFile(
+          target,
+          new Command(
+            `compile job -s ${process.projectDir}/jobs/extract/makedata1.sas`
+          ),
+          'job'
+        )
+      ).toResolve()
+      expect(compileJobFile.compileJobFile).toHaveBeenCalled()
+
+      done()
+    })
+  })
+
+  describe('service', () => {
+    beforeEach(async (done) => {
+      appName = `cli-tests-compile-${generateTimestamp()}`
+      await createTestApp(__dirname, appName)
+      jest.spyOn(compileServiceFile, 'compileServiceFile')
+      done()
+    })
+
+    it('should compile single file', async (done) => {
+      await expect(
+        compileSingleFile(
+          target,
+          new Command(
+            `compile service -s ${process.projectDir}/sasjs/services/common/example.sas`
+          ),
+          'service'
+        )
+      ).toResolve()
+      expect(compileServiceFile.compileServiceFile).toHaveBeenCalled()
```

Comment:
The fact that `compileJobFile` has been called doesn't prove that a single file has been compiled properly and saved at the correct location.

Reply:
Added `compileServiceFile` tests separately.